src/app/components/section/section.component.ts

```diff
@@ -32,8 +35,13 @@ export class SectionComponent implements OnInit {
     console.log(text);
   }
 
-  public showTheSearching(searchText: string): void {
+  public showTheSearching(searchText: string): any {
     console.log(searchText)
+
+    if (!searchText) {
+      return this.courses = this.courseService.getCourses()
+    }
+    return this.courses = this.courses?.filter(course => course.title.includes(searchText))
```

Comment:
why you filter data if you use filter pipe?

Reply:
cause if i dont use filter data and don't click button, show more doesn't disappear and noCourseFeelFreeToAddData doesn't appear